python/letsql/tests/test_array.py

```diff
@@ -112,3 +126,70 @@ def test_array_remove(con):
     result = con.execute(expr)
     expected = pd.Series([[3], [], [42], [], []], dtype="object")
     assert_series_equal(result, expected, check_names=False)
+
+
+@pytest.mark.parametrize(
+    ("column", "expected"),
+    [
+        param("empty", pd.Series([[], [], []], dtype="object"), id="empty"),
+        param(
+            "happy", pd.Series([["abc"], ["bcd"], ["def"]], dtype="object"), id="happy"
+        ),
+    ],
+)
+def test_array_flatten(con, flatten_data, column, expected):
+    data = flatten_data[column]
+    t = ibis.memtable({column: data["data"]}, schema={column: data["type"]})
+    expr = t[column].flatten()
+    result = con.execute(expr)
+    assert_series_equal(
+        result.sort_values().reset_index(drop=True),
+        expected.sort_values().reset_index(drop=True),
+        check_names=False,
+    )
+
+
+@pytest.mark.parametrize("step", [-2, -1, 1, 2])
+@pytest.mark.parametrize(
+    ("start", "stop"),
+    [
+        param(-7, -7),
+        param(-7, 0),
+        param(-7, 7),
+        param(0, -7),
+        param(0, 0),
+        param(0, 7),
+        param(7, -7),
+        param(7, 0),
+        param(7, 7),
+    ],
+)
+def test_range_start_stop_step(con, start, stop, step):
+    expr = ibis.range(start, stop, step)
+    result = con.execute(expr)
+    assert list(result) == list(range(start, stop, step))
+
+
+@pytest.mark.parametrize("n", [-2, 0, 2])
+def test_range_single_argument(con, n):
+    expr = ibis.range(n)
+    result = con.execute(expr)
+    assert list(result) == list(range(n))
+
+
+@pytest.mark.parametrize(
+    ("data", "expected"),
+    [
+        param(
+            {"a": [[1, 3, 3], [], [42, 42], []]},
+            [{3.0, 1.0}, set(), {42.0}, set()],
+            id="not_null",
+        ),
+    ],
+)
+def test_array_unique(con, data, expected):
+    t = ibis.memtable(data)
+    expr = t.a.unique()
+    result = con.execute(expr).map(frozenset, na_action="ignore")
+    expected = pd.Series(expected, dtype="object").map(frozenset, na_action="ignore")
+    assert frozenset(result.values) == frozenset(expected.values)
```

Comment:
why do we use frozenset here?
could we test against the `pd.Series` result of `execute` directly?
```
In [20]: expr.execute().equals(pd.Series([[1, 3], [], [42], []]))
Out[20]: True
```

Reply:
why does `expected` contain `float`s rather than `int`s?

Reply:
Fixed: I'm using `assert_series_equal`